src/Bicep.LangServer/Handlers/BicepTextDocumentSyncHandler.cs

```diff
@@ -34,30 +48,68 @@ public override Task<Unit> Handle(DidChangeTextDocumentParams request, Cancellat
             // we have full sync enabled, so apparently first change is the whole document
             var contents = request.ContentChanges.First().Text;
 
-            this.compilationManager.UpsertCompilation(request.TextDocument.Uri, request.TextDocument.Version, contents);
+            var documentUri = request.TextDocument.Uri;
+
+            this.compilationManager.UpsertCompilation(documentUri, request.TextDocument.Version, contents);
+
+            if (IsBicepConfigFile(documentUri))
+            {
+                var configuration = configurationManager.GetConfiguration(documentUri.ToUri());
+                activeBicepConfigCache.AddOrUpdate(documentUri, (documentUri) => configuration, (documentUri, prevConfiguration) => configuration);
+            }
 
             return Unit.Task;
         }
 
         public override Task<Unit> Handle(DidOpenTextDocumentParams request, CancellationToken cancellationToken)
         {
-            this.compilationManager.UpsertCompilation(request.TextDocument.Uri, request.TextDocument.Version, request.TextDocument.Text, request.TextDocument.LanguageId);
+            var documentUri = request.TextDocument.Uri;
+
+            if (IsBicepConfigFile(documentUri))
+            {
+                var configuration = configurationManager.GetConfiguration(documentUri.ToUri());
+                activeBicepConfigCache.TryAdd(documentUri, configuration);
+            }
+
+            this.compilationManager.UpsertCompilation(documentUri, request.TextDocument.Version, request.TextDocument.Text, request.TextDocument.LanguageId);
 
             return Unit.Task;
         }
 
         public override Task<Unit> Handle(DidSaveTextDocumentParams request, CancellationToken cancellationToken)
         {
+            var documentUri = request.TextDocument.Uri;
+
+            if (IsBicepConfigFile(documentUri) &&
+                activeBicepConfigCache.TryRemove(documentUri, out RootConfiguration? prevBicepConfiguration) &&
+                prevBicepConfiguration != null)
+            {
+                var curConfiguration = configurationManager.GetConfiguration(documentUri.ToUri());
+                TelemetryHelper.SendTelemetryOnBicepConfigChange(prevBicepConfiguration, curConfiguration, linterRulesProvider, telemetryProvider);
+            }
+
             // nothing needs to be done when the document is saved
```

Comment:
Guess this comment is no longer accurate

Reply:
Removed here:
Added error handling to not block compilation:
https://github.com/Azure/bicep/commit/979995aec3298bb62719b50267e746b3fd667ffb